doc/uftrace-test.md

```diff
@@ -0,0 +1,333 @@
+test command: ../uftrace --no-pager --no-event -L.. t-abc
+
+````
+
+``````
+root@ubuntu:~/uftrace/tests$ ./runtest.py 001 -v -O1
+Test case                 pg fi
+------------------------: O1 O1
+build command: gcc -o t-abc -fno-inline -fno-builtin -fno-ipa-cp -fno-omit-frame-pointer -D_FORTIFY_SOURCE=0  -pg -O1  s-abc.c   
+test command: ../uftrace --no-pager --no-event -L.. t-abc
+``````
+
+`````
+root@ubuntu:~/uftrace/tests$ ./runtest.py 001 -i -v -O1
+Test case                 fi
+------------------------: O1
+build command: gcc -o t-abc -fno-inline -fno-builtin -fno-ipa-cp -fno-omit-frame-pointer -D_FORTIFY_SOURCE=0  -finstrument-functions -O1  s-abc.c   
+test command: ../uftrace --no-pager --no-event -L.. t-abc
+`````
+
+`````
+root@ubuntu:~/uftrace/tests$ ./runtest.py 001 -p -v -O1
+Test case                 pg
+------------------------: O1
+build command: gcc -o t-abc -fno-inline -fno-builtin -fno-ipa-cp -fno-omit-frame-pointer -D_FORTIFY_SOURCE=0  -pg -O1  s-abc.c   
+test command: ../uftrace --no-pager --no-event -L.. t-abc
+
+`````
+
+## ADD TEST CASE
+
+If you do not have the test code you want, you can write the test code and conduct the test.
+
+- Example To add the desired test case:
+
+**first > create s-[TEST CASE NAME].c / s-[TEST CASE NAME].cpp (In your preferred language)**
+
+```
+root@ubuntu:~/uftrace/tests$ cat s-USER_TEST.c 
+#include<stdio.h>
+
+void TEST(void);
+
+int main()
+{
+	TEST();
+
+	return 0;
+}
+
+void TEST(void){
+	printf("USER_TEST_CODE\n");
+}
+```
+
+**second > Apply the default gcc and uftrace options in runtest.py to get the following results:**
+(If you want, you can optionally modify your runtest.py to give your own default gcc option or uftrace.)
+
+```
+root@ubuntu:~/uftrace/tests$ gcc -o t-USER_TEST -fno-inline -fno-builtin -fno-ipa-cp -fno-omit-frame-pointer -D_FORTIFY_SOURCE=0  -finstrument-functions -Os  s-USER_TEST.c
+<command-line>:0:0: warning: "_FORTIFY_SOURCE" redefined
+<built-in>: note: this is the location of the previous definition
+
+root@ubuntu:~/uftrace/tests$ uftrace --no-pager --no-event -L.. t-USER_TEST
+USER_TEST_CODE
+# DURATION     TID     FUNCTION
+            [  6053] | main() {
+            [  6053] |   TEST() {
+ 149.720 us [  6053] |     printf();
+ 151.607 us [  6053] |   } /* TEST */
+ 152.396 us [  6053] | } /
+```
+
+**Third > create t[TEST CASE NUMBER]_[TEST CASE NAME].py and paste second result**
+
+```
+#!/usr/bin/env python
+
+from runtest import TestBase
+
+class TestCase(TestBase):
+    def __init__(self):
+        TestBase.__init__(self, 'USER_TEST', """
+# DURATION     TID     FUNCTION
+   0.858 us [  6053] | __monstartup();
+   0.332 us [  6053] | __cxa_atexit();
+            [  6053] | main() {
+            [  6053] |   TEST() {
+ 131.151 us [  6053] |     printf();
+ 132.454 us [  6053] |   } /* TEST */
+ 132.658 us [  6053] | } /* main */
+ """)
+
+        def runcmd(self):
+            return '%s -N c %s' % (TestBase.uftrace_cmd, 't-USER_TEST')
+```
```

Comment:
At the minimum, you don't even need the `runcmd()` function.  The runtest.py will run uftrace based on your test name (given on `__init__()`) with no option.